lib/collision_detection.dart

```diff
@@ -0,0 +1,32 @@
+import 'dart:math' as math;
+
+import 'extensions/vector2.dart';
+
+/// Checks whether the [polygon] represented by the list of [Vector2] contains
+/// the [point].
+bool containsPoint(Vector2 point, List<Vector2> polygon) {
+  for (int i = 0; i < polygon.length; i++) {
+    final previousNode = polygon[i];
+    final node = polygon[(i + 1) % polygon.length];
+    final isOutside = (node.x - previousNode.x) * (point.y - previousNode.y) -
+            (point.x - previousNode.x) * (node.y - previousNode.y) >
+        0;
+    if (isOutside) {
+      // Point is outside of convex polygon
+      return false;
+    }
+  }
+  return true;
+}
+
+// Rotates the [point] with [angle] around [position]
+Vector2 rotatePoint(Vector2 point, double angle, Vector2 position) {
```

Comment:
The old Position class had this method, don't we have an extension function on Vector2 (or something native) to do this? if so we should add instead of an utils

Reply:
Makes sense, migrated this to the `Vector2` extension.